x/ibc/02-client/handler.go

```diff
@@ -104,3 +106,48 @@ func HandlerClientMisbehaviour(k keeper.Keeper) evidencetypes.Handler {
 		return nil
 	}
 }
+
+// HandleClientUpdateProposal is a handler for executing a passed client update proposal
+func HandleClientUpdateProposal(ctx sdk.Context, k keeper.Keeper, p *ibctypes.ClientUpdateProposal) error {
+	clientState, found := k.GetClientState(ctx, p.ClientId)
+	if !found {
+		return types.ErrClientNotFound
+	}
+
+	clientType := clientState.ClientType()
+	switch clientType {
```

Comment:
this is not scalable as we don't know how many clients will be supported in the future. We should call a function from the client state interface in order to support all the clients

Reply:
So client interface will be something like this:

![image](https://user-images.githubusercontent.com/30392990/90701025-9f579180-e287-11ea-9867-6226ffa94969.png)

I plan to add also `GetLatestTimestamp`. Is it ok?